botocore/utils.py

```diff
@@ -1569,6 +1571,45 @@ def set_endpoint(self, request, **kwargs):
     def _use_accesspoint_endpoint(self, request):
         return 's3_accesspoint' in request.context
 
+    def _validate_fips_supported(self, request):
+        client_region = request.context.get('client_region', '')
+        if 'fips' not in client_region:
+            return
+        if 'outpost_name' in request.context['s3_accesspoint']:
+            raise UnsupportedS3AccesspointConfigurationError(
+                msg=(
+                    'Invalid configuration Outpost Access Points '
+                    'do not support Fips- regions'
+                )
+            )
+        client_region = client_region.replace('fips-', '')
+        if request.context['s3_accesspoint']['region'] != client_region:
+            if self._s3_config.get('use_arn_region'):
+                raise UnsupportedS3AccesspointConfigurationError(
+                    msg=(
+                        'Invalid configuration, '
+                        'FIPS region does not match ARN region'
+                    )
+                )
```

Comment:
Should we be erroring out when the region does not match but you opted into using the arn region? The general behavior looks like it has always been just use the region from the arn unless it is outside the partition of the configured client: https://github.com/boto/botocore/blob/92c63ea94542c5d54d582c50509be02b96d7a443/botocore/utils.py#L1580-L1589